src/Umbraco.Tests.Common/Builders/TemplateBuilder.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using Umbraco.Core.Models;
+using Umbraco.Core.Strings;
+using Umbraco.Tests.Common.Builders.Extensions;
+using Umbraco.Tests.Common.Builders.Interfaces;
+
+namespace Umbraco.Tests.Common.Builders
+{
+    public class TemplateBuilder
+        : BuilderBase<Template>,
+            IWithIdBuilder,
+            IWithKeyBuilder,
+            IWithAliasBuilder,
+            IWithNameBuilder,
+            IWithCreateDateBuilder,
+            IWithUpdateDateBuilder,
+            IWithPathBuilder
+    {
+        private int? _id;
+        private Guid? _key;
+        private string _alias;
+        private string _name;
+        private DateTime? _createDate;
+        private DateTime? _updateDate;
+        private string _path;
+        private string _content;
+        private bool? _isMasterTemplate;
+        private string _masterTemplateAlias;
+        private Lazy<int> _masterTemplateId;
+
+        public TemplateBuilder WithContent(string content)
+        {
+            _content = content;
+            return this;
+        }
+
+        public TemplateBuilder AsMasterTemplate(string masterTemplateAlias, int masterTemplateId)
+        {
+            _isMasterTemplate = true;
+            _masterTemplateAlias = masterTemplateAlias;
+            _masterTemplateId = new Lazy<int>(() => masterTemplateId);
+            return this;
+        }
+
+        public override Template Build()
+        {
+            var id = _id ?? 0;
+            var key = _key ?? Guid.NewGuid();
+            var name = _name ?? Guid.NewGuid().ToString();
+            var alias = _alias ?? name.ToCamelCase();
+            var createDate = _createDate ?? DateTime.Now;
+            var updateDate = _updateDate ?? DateTime.Now;
+            var path = _path ?? string.Empty;
+            var content = _content;
```

Comment:
Why are we skipping the null check here?
```suggestion
            var content = _content ?? string.Empty;
```

Reply:
I found we needed to retain this as null, otherwise we'd get a test failure here; `Assert.AreEqual(clone, template);`.  Seems null specifically is expected here, rather than empty string.